opengrok-indexer/src/main/java/org/opengrok/indexer/web/messages/Message.java

```diff
@@ -50,7 +50,44 @@
     @NotEmpty(message = "tags cannot be empty")
     private Set<String> tags = Collections.singleton(MESSAGES_MAIN_PAGE_TAG);
 
-    private String cssClass = "info";
+    public enum CssClassType {
+        /**
+         * Known values: SUCCESS, INFO, WARNING, ERROR.
+         * The values are sorted according to their level. Higher numeric value of the level (i.e. the enum ordinal)
+         * means higher priority.
+         */
+        SUCCESS("success"), INFO("info"), WARNING("warning"), ERROR("error");
+
+        private String cssClassString;
+
+        CssClassType(String str) {
+            cssClassString = str;
+        }
+
+        public static CssClassType StringToCssClassType(String val) throws IllegalArgumentException {
+            if (val == null) {
+                return INFO;
+            }
+            for (CssClassType v : CssClassType.values()) {
+                if (v.toString().equals(val)) {
+                    return v;
+                }
+            }
+            throw new IllegalArgumentException("class type does not match any known value");
+        }
+
+        @Override
+        public String toString() {
+            return cssClassString;
+        }
+
+        public static int compare(CssClassType c1, CssClassType c2) {
```

Comment:
there is compareTo of a Comparable interface

Reply:
you should implement it

Reply:
This is needed for `max()` which requires `Comparator` rather than `Comparable`. However, the method needs to be static so it can be used via method reference hence the class cannot implement `Comparator`.

Reply:
perhaps using `Comparator` as a static member is the cleanest way.